src/util/sss_ssh.c

```diff
@@ -18,13 +18,59 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <ctype.h>
 #include <arpa/inet.h>
 
 #include "db/sysdb.h"
 #include "util/util.h"
 #include "util/crypto/sss_crypto.h"
 #include "util/sss_ssh.h"
 
+
+#define SKIP_SPACES(p)       while (*p != '\0' &&  isspace(*p)) { p++; }
+#define SKIP_UNTIL_SPACE(p)  while (*p != '\0' && !isspace(*p)) { p++; }
+
+/* Check if the key_line we received from the backend includes the hostname or
+ * it is just the keytype and the key.
+ *
+ * Key lines have this format:
+ * marker (optional), hostnames, keytype, base64-encoded key, comment (optional)
+ *
+ * This is a very simplistic method based on looking for the provided hostname
+ * into the provided keyline, at the right position (the hostname could also
+ * be present in the comment at the end).
+ */
+static bool
+sss_ssh_key_has_host_name(const char *key_line, const char *hostname)
+{
+    const char *current = (const char *) key_line;
+    const char *end;
+    size_t len;
+
+    SKIP_SPACES(current);
```

Comment:
I would just inline.
If you think this helps readability, then simply add /* skip spaces */ comment.

Writing a safe macro is tricky.
If possible, better skip.
Even if it's currently used in one place in a safe way, it doesn't mean nobody will be tempter to re-use it later in unsafe way (for example: `SKIP_SPACES(p++)`)

Reply:
I will inline them. I initially thought they would be used more than once.